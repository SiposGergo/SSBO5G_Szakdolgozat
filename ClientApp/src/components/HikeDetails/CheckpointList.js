import React from "react";
import CheckpointListRow from "./ChecpointListRow";

const CheckpointList = (props) => (
    <div>
        <table>
            <tbody>
                <tr>
                    <th>Név</th>
                    <th>Nyitás</th>
                    <th>Zárás</th>
                    <th>Távolság a rajttól</th>
                    <th>Leírás</th>
                </tr>
                {props.checkpoints.map((cp) =>
                    (<CheckpointListRow checkpoint={cp} key={cp.id} />))
                }
            </tbody>
        </table>
    </div>
)

export default CheckpointList;
