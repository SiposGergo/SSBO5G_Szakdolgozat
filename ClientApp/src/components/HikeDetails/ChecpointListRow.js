import React from "react";
import moment from "moment";

const CheckpointListRow = (props) => (
    <tr>
        <td>{props.checkpoint.name}</td>
        <td>{moment(props.checkpoint.open).format('HH:mm')}</td>
        <td>{moment(props.checkpoint.close).format('HH:mm')}</td>
        <td>{props.checkpoint.distanceFromStart/1000} km</td>
        <td>{props.checkpoint.description}</td>
    </tr>
)

export default CheckpointListRow;