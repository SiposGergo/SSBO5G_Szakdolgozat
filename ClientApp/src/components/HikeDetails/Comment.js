import React from "react";
import moment from "moment";

class Comment extends React.Component {
    render() {
        const comment = this.props.comment;
        return (
            <div>
                <p>{moment(comment.timeStamp).format('YYYY MM DD HH:mm')}</p>
                <p>{comment.author.name} : {comment.commentText}</p>
            </div>
        )
    }
}

export default Comment;