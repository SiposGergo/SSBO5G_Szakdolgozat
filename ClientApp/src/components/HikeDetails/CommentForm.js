import React from 'react'
import { Field, reduxForm } from 'redux-form'

const CommentForm = props => {
  const { handleSubmit, pristine, reset, submitting } = props
  return (
    <form onSubmit={handleSubmit}>
    <p>Írj kommentet!</p>
    <div>
    <div>
      <Field
        name="message"
        component="textarea"
        placeholder="Üzenet..."
        rows = "8"
        cols = "50"
      />
    </div>
    </div>
      <div>
        <button type="submit" disabled={pristine || submitting}>
          Elküld
        </button>
      </div>
    </form>
  )
}

export default reduxForm({
  form: 'CommentForm'
})(CommentForm)