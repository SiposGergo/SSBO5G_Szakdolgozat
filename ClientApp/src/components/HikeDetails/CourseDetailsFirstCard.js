import React from "react";
import moment from "moment";
import {config} from "../../helpers/config";

class CourseDetailsFirstCard extends React.Component {
    render() {
        const props = this.props;
        const registratedPercent = ((props.course.numOfRegisteredHikers / props.course.maxNumOfHikers) * 100).toString() + "%";
        
        return (
            <div>
                <p>{props.course.name}</p>
                <p>Nevezési díj:{props.course.price}</p>
                <p>Táv:{props.course.distance}</p>
                <p>Szint emelkedés:{props.course.elevation}</p>
                <p>Rajt:{props.course.placeOfStart}</p>
                <p>Cél:{props.course.placeOfFinish}</p>
                <p>Rajt ideje:{moment(props.course.beginningOfStart).format(config.dateTimeFormat)}
                    - {moment(props.course.endOfStart).format(config.dateTimeFormat)}</p>
                <p>Szintidő:{props.course.limitTime}</p>
                <p>Létszámkorlát:{props.course.maxNumOfHikers}</p>
                <p>{props.course.numOfRegisteredHikers}/{props.course.maxNumOfHikers}</p>
                <div className="progress">
                    <div className="progress-bar" style={{ width: registratedPercent }}></div>

                </div>
                <p>Nevezési határidő: :{moment(props.course.registerDeadline).format(config.dateTimeFormat)}</p>

            </div>
        )
    }
}

export default CourseDetailsFirstCard;