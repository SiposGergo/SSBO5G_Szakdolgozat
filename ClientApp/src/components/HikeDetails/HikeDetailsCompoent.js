import React from "react";
import moment from "moment";
import {config} from "../../helpers/config.js";


class HikeDetailsCompoent extends React.Component {
    render() {
        const hike = this.props.hike;
        return (
            <div className="col-md details-box">
                <h3 className="flo">{moment(hike.date).format(config.dateFormat)}</h3>
                <h3>{hike.name}</h3>
                <div>{hike.description}</div>
                <a href={hike.website} target="_blank"><button className="btn btn-green">A TÃºra oldala</button></a>
            </div>
        );
    }
}

export default HikeDetailsCompoent;