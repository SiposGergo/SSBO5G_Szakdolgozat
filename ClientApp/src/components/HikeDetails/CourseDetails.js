import React from "react";
import moment from "moment";
import CheckpointList from "./CheckpointList";

class CourseDetails extends React.Component {
    render() {
        const course = this.props.course;
        return (
            <div>
                <p>{course.name}</p>
                <p>Nevezési díj:{course.price}</p>
                <p>Táv:{course.distance}</p>
                <p>Szint emelkedés:{course.elevation}</p>
                <p>Rajt:{course.placeOfStart}</p>
                <p>Cél:{course.placeOfFinish}</p>
                <p>Rajt ideje:{moment(course.beginningOfStart).format('YYYY MM DD HH:mm')}
                    - {moment(course.endOfStart).format('YYYY MM DD HH:mm')}</p>
                <p>Szintidő:{course.limitTime}</p>
                <CheckpointList checkpoints={course.checkPoints} />
            </div>
        )
    }
}

export default CourseDetails;