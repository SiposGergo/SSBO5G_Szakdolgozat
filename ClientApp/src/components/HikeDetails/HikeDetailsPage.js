import React from "react";
import ReactLoading from "react-loading";
import HikeDetailsCompoent from "./HikeDetailsCompoent";
import OrganizerDetailsComponent from "./OrganizerDetailsComponent";
import Comment from "./Comment";
import CourseDetails from "./CourseDetails";
import { Tab } from 'semantic-ui-react';
import CommentForm from "./CommentForm";
import { connect } from 'react-redux';
import { getHikeDetails, postComent } from "../../actions/HikeDetailsActions";
import { SendDanger } from "../../services/NotificationSender";
import { change } from 'redux-form';


export class HikeDetailsPage extends React.Component {

    componentDidMount() {
        this.props.dispatch(getHikeDetails(this.props.match.params.id));
    }

    submitComment = (value) => {
        if (!this.props.user) {
            this.props.dispatch(SendDanger("Lépj be a kommenteléshez!"));
        } else {
            this.props.dispatch(postComent(this.props.hike.id, this.props.user.id, value.message));
            this.props.dispatch(change('CommentForm', 'message', ''));
        }
    }

    render() {
        if (this.props.hasErrored) {
            return <p>Hiba!</p>;
        }

        if (this.props.isLoading) {
            return <ReactLoading type="spin" color="#000000" height={40} width={40} />
        }

        const hike = this.props.hike;

        const panes = hike.courses
            .map((course) => {
                return {
                    menuItem: course.name,
                    render: () => (<Tab.Pane><CourseDetails hikeId={hike.id} key={course.id} course={course} /></Tab.Pane>)
                }
            });

        return (
            <div style={{width:"90%"}}>
                <HikeDetailsCompoent hike={hike} />
                <OrganizerDetailsComponent organizer={hike.organizer} />
                <div >
                    <Tab menu={{ color:"green", size:"huge", width:"2", inverted: true, attached: true }} panes={panes} />
                </div>
                <div>
                {
                    hike.comments.length == 0 ?
                        <p>Nincsenek hozzászólások</p> :
                        hike.comments.map((comment) => <Comment key={comment.id} comment={comment}  user={this.props.user}/>)
                }
                </div>
                <CommentForm onSubmit={this.submitComment} />
            </div>
        )
    }
}

function mapStateToProps(state) {
    const { hasErrored, isLoading, hike } = state.hikeDetailsReducer;
    return {
        hasErrored,
        isLoading,
        hike,
        user: state.authentication.user
    };
}

export default connect(mapStateToProps, null, null, { pure: false })(HikeDetailsPage);