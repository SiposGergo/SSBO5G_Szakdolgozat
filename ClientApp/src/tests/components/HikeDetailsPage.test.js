import React from "react";
import { shallow } from "enzyme";
import {HikeDetailsPage} from "../../components/HikeDetails/HikeDetailsPage";
import {testHikeDetails} from "../fixtures/testHikeDetails"

test('Should render HikeDetails page when error', () => {
    const wrapper = shallow(<HikeDetailsPage 
        hasErrored={true} 
        dispatch={()=>{}}
        match={{params:{id:2}}} 
        />);
    expect(wrapper).toMatchSnapshot();
});

test('Should render HikeDetails page when loading', () => {
    const wrapper = shallow(<HikeDetailsPage 
        isLoading={true} 
        dispatch={()=>{}}
        match={{params:{id:2}}} 
        />);
    expect(wrapper).toMatchSnapshot();
});

test('Should render HikeDetails page with test hike', () => {
    const wrapper = shallow(<HikeDetailsPage 
        hike={testHikeDetails} 
        dispatch={()=>{}}
        match={{params:{id:2}}} 
        />);
    expect(wrapper).toMatchSnapshot();
});