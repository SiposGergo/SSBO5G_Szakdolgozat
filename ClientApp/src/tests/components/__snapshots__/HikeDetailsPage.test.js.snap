// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render HikeDetails page when error 1`] = `
<p>
  Hiba!
</p>
`;

exports[`Should render HikeDetails page when loading 1`] = `
<Loading
  color="#000000"
  delay={1000}
  height={40}
  type="spin"
  width={40}
/>
`;

exports[`Should render HikeDetails page with test hike 1`] = `
<div>
  <HikeDetailsCompoent
    hike={
      Object {
        "comments": Array [
          Object {
            "author": Object {
              "dateOfBirth": "1997-01-01T00:00:00",
              "email": "asd@bme.com",
              "gender": "Male",
              "id": 2,
              "name": "Laci",
              "password": null,
              "phoneNumber": null,
              "town": "Nagyréde",
              "userName": "laci01",
            },
            "authorId": 2,
            "commentText": "trolololoolo",
            "hikeId": 2,
            "id": 4,
            "timeStamp": "2018-05-26T20:56:55.6583397+02:00",
          },
        ],
        "courses": Array [
          Object {
            "beginningOfStart": "2018-07-22T06:00:00",
            "checkPoints": Array [
              Object {
                "close": "2018-07-22T07:00:00",
                "description": "",
                "distanceFromStart": 0,
                "id": 13,
                "name": "Rajt",
                "open": "2018-07-22T06:00:00",
              },
              Object {
                "close": "2018-07-22T07:40:00",
                "description": "",
                "distanceFromStart": 2600,
                "id": 14,
                "name": "Kecske kő",
                "open": "2018-07-22T06:10:00",
              },
              Object {
                "close": "2018-07-22T08:30:00",
                "description": "",
                "distanceFromStart": 5800,
                "id": 15,
                "name": "Várhegy",
                "open": "2018-07-22T06:20:00",
              },
              Object {
                "close": "2018-07-22T09:15:00",
                "description": "",
                "distanceFromStart": 8900,
                "id": 16,
                "name": "Diós-kút",
                "open": "2018-07-22T06:50:00",
              },
              Object {
                "close": "2018-07-22T09:40:00",
                "description": "",
                "distanceFromStart": 11100,
                "id": 17,
                "name": "Fajzatpuszta",
                "open": "2018-07-22T07:00:00",
              },
              Object {
                "close": "2018-07-22T10:20:00",
                "description": "",
                "distanceFromStart": 13300,
                "id": 18,
                "name": "Káva",
                "open": "2018-07-22T07:05:00",
              },
              Object {
                "close": "2018-07-22T11:20:00",
                "description": "",
                "distanceFromStart": 16900,
                "id": 19,
                "name": "Tót-hegyes",
                "open": "2018-07-22T07:25:00",
              },
              Object {
                "close": "2018-07-22T12:10:00",
                "description": "",
                "distanceFromStart": 20100,
                "id": 20,
                "name": "Hideg-kút",
                "open": "2018-07-22T07:40:00",
              },
              Object {
                "close": "2018-07-22T13:15:00",
                "description": "kilátó",
                "distanceFromStart": 24500,
                "id": 21,
                "name": "Nyikom",
                "open": "2018-07-22T08:00:00",
              },
              Object {
                "close": "2018-07-22T14:30:00",
                "description": "",
                "distanceFromStart": 28700,
                "id": 22,
                "name": "Mátrakeresztes",
                "open": "2018-07-22T08:40:00",
              },
              Object {
                "close": "2018-07-22T16:20:00",
                "description": "",
                "distanceFromStart": 36500,
                "id": 23,
                "name": "Világos-hegy",
                "open": "2018-07-22T08:50:00",
              },
              Object {
                "close": "2018-07-22T17:00:00",
                "description": "",
                "distanceFromStart": 39200,
                "id": 24,
                "name": "Nagy-lénia",
                "open": "2018-07-22T09:00:00",
              },
              Object {
                "close": "2018-07-22T17:30:00",
                "description": "",
                "distanceFromStart": 41200,
                "id": 25,
                "name": "Fajzatpuszta",
                "open": "2018-07-22T09:10:00",
              },
              Object {
                "close": "2018-07-22T18:15:00",
                "description": "",
                "distanceFromStart": 44200,
                "id": 26,
                "name": "Havas",
                "open": "2018-07-22T09:30:00",
              },
              Object {
                "close": "2018-07-22T18:50:00",
                "description": "",
                "distanceFromStart": 46400,
                "id": 27,
                "name": "Puskaporos-kút",
                "open": "2018-07-22T09:50:00",
              },
              Object {
                "close": "2018-07-22T20:00:00",
                "description": "",
                "distanceFromStart": 51400,
                "id": 28,
                "name": "Cél",
                "open": "2018-07-22T10:00:00",
              },
            ],
            "distance": 51400,
            "elevation": 1992,
            "endOfStart": "2018-07-22T08:00:00",
            "hikeId": 2,
            "id": 2,
            "limitTime": "12:00:00",
            "maxNumOfHikers": 60,
            "name": "Patai Mátra 50",
            "placeOfFinish": "Gyöngyöspata, általános iskola",
            "placeOfStart": "Gyöngyöspata, általános iskola",
            "price": 1600,
            "registerDeadline": "2018-07-21T00:00:00",
          },
          Object {
            "beginningOfStart": "2018-07-22T06:00:00",
            "checkPoints": Array [],
            "distance": 34400,
            "elevation": 1338,
            "endOfStart": "2018-07-22T08:00:00",
            "hikeId": 2,
            "id": 3,
            "limitTime": "10:00:00",
            "maxNumOfHikers": 100,
            "name": "Patai Mátra 35",
            "placeOfFinish": "Gyöngyöspata, általános iskola",
            "placeOfStart": "Gyöngyöspata, általános iskola",
            "price": 1400,
            "registerDeadline": "2018-07-21T00:00:00",
          },
        ],
        "date": "2018-07-22T00:00:00",
        "description": "Patai Mátra telejsítménytúra. Túra Gyöngyöspatáról 4 távon.",
        "id": 2,
        "name": "Patai Mátra",
        "organizer": Object {
          "dateOfBirth": "1995-01-01T00:00:00",
          "email": "asd@alfold.com",
          "gender": "Male",
          "id": 4,
          "name": "Erdei László",
          "password": null,
          "phoneNumber": "06303084417",
          "town": "Nagyréde",
          "userName": "laci02",
        },
        "website": "http://www.alfoldte.hu",
      }
    }
  />
  <OrganizerDetailsComponent
    organizer={
      Object {
        "dateOfBirth": "1995-01-01T00:00:00",
        "email": "asd@alfold.com",
        "gender": "Male",
        "id": 4,
        "name": "Erdei László",
        "password": null,
        "phoneNumber": "06303084417",
        "town": "Nagyréde",
        "userName": "laci02",
      }
    }
  />
  <div>
    <Tab
      grid={
        Object {
          "paneWidth": 12,
          "tabWidth": 4,
        }
      }
      menu={
        Object {
          "aligned": "left",
          "attached": true,
          "tabular": true,
        }
      }
      panes={
        Array [
          Object {
            "menuItem": "Patai Mátra 50",
            "render": [Function],
          },
          Object {
            "menuItem": "Patai Mátra 35",
            "render": [Function],
          },
        ]
      }
      renderActiveOnly={true}
    />
  </div>
  <Comment
    comment={
      Object {
        "author": Object {
          "dateOfBirth": "1997-01-01T00:00:00",
          "email": "asd@bme.com",
          "gender": "Male",
          "id": 2,
          "name": "Laci",
          "password": null,
          "phoneNumber": null,
          "town": "Nagyréde",
          "userName": "laci01",
        },
        "authorId": 2,
        "commentText": "trolololoolo",
        "hikeId": 2,
        "id": 4,
        "timeStamp": "2018-05-26T20:56:55.6583397+02:00",
      }
    }
    key="4"
  />
  <ReduxForm
    onSubmit={[Function]}
  />
</div>
`;
